react/features/overlay/components/PageReloadOverlay.js

```diff
@@ -0,0 +1,292 @@
+         * @type {boolean}
+         */
+        isNetworkFailure: React.PropTypes.bool,
+
+        /**
+         * The reason for the error that will cause the reload.
+         * NOTE: Used by PageReloadOverlay only.
+         * @public
+         * @type {string}
+         */
+        reason: React.PropTypes.string
+    }
+
+    /**
+     * Initializes a new PageReloadOverlay instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     * @public
+     */
+    constructor(props) {
+        super(props);
+
+        /**
+         * How long the overlay dialog will be
+         * displayed, before the conference will be reloaded.
+         * @type {number}
+         */
+        const timeoutSeconds = 10 + randomInt(0, 20);
+
+        let isLightOverlay, message, title;
+
+        if (this.props.isNetworkFailure) {
+            title = 'dialog.conferenceDisconnectTitle';
+            message = 'dialog.conferenceDisconnectMsg';
+            isLightOverlay = true;
+        } else {
+            title = 'dialog.conferenceReloadTitle';
+            message = 'dialog.conferenceReloadMsg';
+            isLightOverlay = false;
+        }
+
+        this.state = {
+            ...this.state,
+
+            /**
+             * Indicates the css style of the overlay. if true - lighter  and
+             * darker otherwise.
+             * @type {boolean}
+             */
+            isLightOverlay,
+
+            /**
+             * The translation key for the title of the overlay
+             * @type {string}
+             */
+            message,
+
+            /**
+             * How long the overlay dialog will be
+             * displayed, before the conference will be reloaded.
+             * @type {number}
+             */
+            timeoutSeconds,
+
+            /**
+             * The translation key for the title of the overlay
+             * @type {string}
+             */
+            title
+        };
+    }
+
+    /**
+     * Renders the button for relaod the page if necessary.
+     *
+     * @returns {ReactElement|null}
+     * @private
+     */
+    _renderButton() {
+        if (this.props.isNetworkFailure) {
+            const cName = 'button-control button-control_primary '
+                + 'button-control_center';
+            const onClickHandler = this._reconnectNow;
+
+            return (
+                <button
+                    className = { cName }
+                    data-i18n = 'dialog.reconnectNow'
+                    id = 'reconnectNow'
+                    onClick = { onClickHandler } />
+            );
+        }
+
+        return null;
+    }
+
+    /**
+     * Constructs overlay body with the warning message and count down towards
+     * the conference reload.
+     *
+     * @returns {ReactElement|null}
+     * @override
+     * @protected
+     */
+    _renderOverlayContent() {
+        const onFinish = this._reconnectNow;
```

Comment:
Do you mean `onFinish = { this._reconnectNow }`?

It seems that we have a eslint rule that the name of the handler should start with "on".